richie/js/data/course/sideEffects/getList.ts

```diff
@@ -0,0 +1,62 @@
+import { call, put, takeLatest } from 'redux-saga/effects';
+
+import Course from '../../../types/Course';
+import formatQueryString from '../../../utils/http/formatQueryString';
+import { addCourse, CourseListGet, didGetCourseList, failedToGetCourseList } from '../actions';
+
+// Use a polymorphic response object so it can be elegantly consumed through destructuration
+interface Response {
+  error?: string;
+  facets?: {
+    organization: {[organizationId: number]: number};
+    subject: {[subjectId: number]: number};
+  };
+  meta?: {
+    limit: number;
+    offset: number;
+    total_count: number;
+  };
+  objects?: Course[];
+}
+
+// Wrap fetch to handle params, headers, parsing & sane response handling
+// NB: some of this logic should be move in a separate module when we reuse it elsewhere
+export function fetchCourses(params?: CourseListGet['params']): Promise<Response> {
+  return fetch('/api/v1.0/course' + formatQueryString(params), {
```

Comment:
Open question: instead of hard-coding API endpoints, what about listing them somewhere as application settings?